src/tests/system/tests/test_authselect.py

```diff
@@ -0,0 +1,494 @@
+    """
+    :title: SSSD profile is selected and functionally tested
+    :setup:
+        1. Create POSIX user "user-1"
+    :steps:
+        1. Select SSSD profile
+        2. Start SSSD
+        3. Authenticate as "user-1"
+    :expectedresults:
+        1. SSSD profile is selected
+        2. SSSD is started
+        3. Authentication is successful for "user-1"
+    :customerscenario: False
+    :requirement: IDM-SSSD-TC: authselect: Authselect SSSD profile manual Client configuration
+    """
+    provider.user("user-1").add()
+
+    client.authselect.select("sssd")
+    client.sssd.start()
+
+    assert client.auth.ssh.password("user-1", "Secret123")
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_authselect__sssd_profile_enable_and_disable_mkhomedir(client: Client, provider: GenericProvider):
+    """
+    :title: SSSD profile is selected with mkhomedir feature
+    :setup:
+        1. Create POSIX user "user-1"
+    :steps:
+        1. Select SSSD profile with with-mkhomedir
+        2. Start SSSD
+        3. Authenticate as "user-1" and check home directory
+        4. Delete "user-1" home directory
+        5. Disable mkhomedir feature
+        6. Authenticate as "user-1" and check home directory
+    :expectedresults:
+        1. SSSD profile is selected
+        2. SSSD is started
+        3. Authentication is successful for "user-1" and home directory exists
+        4. Home directory is deleted
+        5. Feature is disabled
+        6. Authentication is successful for "user-1" and no home directory exists
+    :customerscenario: False
+    :requirement: True
+    """
+    provider.user("user-1").add(home="/home/user-1")
+
+    client.sssd.start()
+    client.authselect.enable_feature(["with-mkhomedir"])
+
+    assert client.auth.ssh.password("user-1", "Secret123")
+    assert client.fs.exists("/home/user-1")
+
+    client.tools.host.ssh.run("rm -rf /home/user-1")
+    client.authselect.disable_feature(["with-mkhomedir"])
+
+    assert client.auth.ssh.password("user-1", "Secret123")
+    assert not client.fs.exists("/home/user-1")
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_authselect__sssd_profile_enable_and_disable_faillock(client: Client, provider: GenericProvider):
+    """
+    :title: SSSD profile is selected with faillock
+    :title: Minimal profile only requires the following packages to run; sssd-client, sssd-common and sssd-nfs-idmap
+    :setup:
+        1. Create POSIX user "user-1"
+    :steps:
+        1. Select the authselect SSSD profile with-faillock
+        2. Configure pam-faillock and set login attempts to 3
+        3. Start SSSD
+        4. Authenticate as "user-1"
+        5. Authenticate as "user-1" 3 times with an invalid password
+        6. Reset faillock for "user-1"
+        7. Authenticate as "user-1"
+        8. Disable feature pam-faillock
+        9. Authenticate as "user-1" 3 times with an invalid password
+        10. Authenticate as "user-1"
```

Comment:
Do we really need to test these steps in this scenario? Wouldn't it be better to move to its own test case?

If you still want to have the steps here then please update the title so that it explains what you are testing

Reply:
It should remain in the same test case. Otherwise, it will have extra code that I think is necessary.  i.e. 

```
test_feature
    setup feature 
    test feature
    disable feature
    test lack of feature

test_enable_feature
    setup feature
    test feature
test_disable_feature
    setup feature
    disable feature
    test lack of feature
```

Reply:
Ok, now that you changed the title it's fine